src/transformation/visitors/call.ts

```diff
@@ -267,7 +268,9 @@ export const transformCallExpression: FunctionVisitor<ts.CallExpression> = (node
 
     let callPath: lua.Expression;
     let parameters: lua.Expression[];
-    const isContextualCall = isContextualCallExpression(context, signature);
+
+    const isContextualCall = isContextualCallExpression(context, signature, node.getSourceFile());
```

Comment:
I actually don't think this is correct. the sourcefile you need to check is the one where the signature is declared, not the file where the call happens. I'd also like to see this in a test (ie the signature is in a different file to the call). If you need help writing the test ping me in discord.

Reply:
Hello, I'm unsure of how this would really be achieved.
For example in the example code above, the declaration variable here is undefined.
Which makes it a bit hard to fetch the actual source file from the declaration variable.

Reply:
I'm not sure either, I would expect to indeed go via the declaration or symbols somehow. I'm pretty sure it should be possible somehow but not sure how without playing around with the API first

Reply:
I'll see if I can play around a bit with it.
Fairly weird why the declaration would be undefined in the first place though.